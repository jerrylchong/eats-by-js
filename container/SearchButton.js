import React from 'react';
import { StyleSheet, TextInput, View, Image, TouchableOpacity } from 'react-native';

class SearchButton extends React.Component {
    state = {
        searchTerm: ''
    }

    handleSearchTerm = searchTerm => this.setState({searchTerm})

    pressSearch = () => {
        this.setState({searchTerm: ''})
    }

    render() {
        return (
            <View style = {styles.container}>
                <TextInput style = {styles.input}
                           placeholder = "Search"
                           placeholderTextColor = '#646464'
                           onChangeText = {this.handleSearchTerm}
                           value = {this.state.searchTerm} />
                <TouchableOpacity onPress = {this.pressSearch}>
                    <Image style = {styles.image} source ={require('../assets/magnifying_glass.png')} />
                </TouchableOpacity>
            </View>
        )
    }
}

export default SearchButton

const styles = StyleSheet.create({

    container: {
        backgroundColor: '#DBDBDB',
        borderRadius: 20,
        borderWidth: 1,
        borderColor: '#DBDBDB',
        width: 375,
        height: 40,
        paddingHorizontal: 15,
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center'
    },
    input: {
        color: '#646464',
        fontSize: 16
    },
    image: {
        height: 13,
        width: 15
    }
})