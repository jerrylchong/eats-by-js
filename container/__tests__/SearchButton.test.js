import React from 'react';
import renderer from 'react-test-renderer';
import SearchButton from "../SearchButton";
import Enzyme, {shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import {TouchableOpacity} from "react-native";

// snapshot test
test('renders correctly', () => {
    const tree = renderer.create(
        <SearchButton
            searchTerm={''}
            navigation={{navigate: jest.fn(), addListener: jest.fn()}}
        />).toJSON();
    expect(tree).toMatchSnapshot();
})

Enzyme.configure({ adapter: new Adapter() });

test('test pressSearch', () => {
    const wrapper = shallow(<SearchButton
        navigation={{navigate: jest.fn(), addListener: jest.fn()}}
        searchTerm={''}
    />)
    wrapper.find(TouchableOpacity).props().onPress(); // press magnifying glass button
    expect(wrapper.state().pressed).toEqual(true); // expect pressed state to be true
})

test('test clearSearch', () => {
    const onPressEvent = jest.fn();
    onPressEvent.mockReturnValue('onPress invoked'); // create mock function
    const wrapper = shallow(<SearchButton
        navigation={{navigate: jest.fn(), addListener: jest.fn()}}
        searchTerm={'a'}
        clearSearch={onPressEvent} // pass mock function as clearSearch prop
    />)
    wrapper.setState({pressed: true});
    wrapper.childAt(1).find(TouchableOpacity).props().onPress(); // press clear search button
    expect(onPressEvent.mock.calls.length).toBe(1); // expect to have 1 mock function call
})