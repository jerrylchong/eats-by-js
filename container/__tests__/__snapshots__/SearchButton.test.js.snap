// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "width": "95%",
    }
  }
>
  <View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 75,
          "justifyContent": "center",
          "opacity": 1,
          "width": 75,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 13,
            "width": 13,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#ececec",
        "borderRadius": 25,
        "height": 90,
        "justifyContent": "center",
        "width": 90,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 90,
          "width": 90,
        }
      }
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "height": 20,
        "opacity": 1,
        "width": 24,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 20,
          "width": 24,
        }
      }
    />
  </View>
</View>
`;
